### DISCUSSION TITLE
Issue: Could commit but not push

### DISCUSSION THREAD

**Discussion by adsmithca** [ateucher/pkg-dev-dfo-2024-11#35] [General]
Hey, 

So yesterday I was able to commit changes but not push them to a github repository. 
I think this happened because the script I was using to run the development code was in the package R directory and was causing havoc with my PAT credentials and the setting of the remote. I thought I had included the setup script in the Rbuildignore file but I may not have done that correctly. 

I decided to start over from scratch. 
After running through the code to create the package and setup credentials I ran 
git_sitrep() I get this: 

 Git global (user) 
• Name: "Andrew Smith"
• Email: "andrew.d.smith.ca@gmail.com"
• Global (user-level) gitignore file: C:/Users/SmithAND/.gitignore
• Vaccinated: TRUE
• Default Git protocol: "https"
• Default initial branch name: "master"

── GitHub user 
• Default GitHub host: "https://github.com"
• Personal access token for "https://github.com": <discovered>
• GitHub user: "adsmithca"
• Token scopes: "gist", "repo", "user", and "workflow"
• Email(s): "andrew.d.smith.ca@gmail.com (primary)" and "47754057+adsmithca@users.noreply.github.com"

── Active usethis project: "C:/Users/SmithAND/Desktop/libminer" ──

── Git local (project) 
• Name: "Andrew Smith"
• Email: "andrew.d.smith.ca@gmail.com"
• Default branch: "main"
• Current local branch → remote tracking branch:
  "main" → <unset>

── GitHub project 
• Type = "ours"
• Host = "https://github.com"
• Config supports a pull request = TRUE
• origin = "adsmithca/libminer" (can push)
• upstream = <not configured>
ℹ "origin" is both the source and primary repo.
ℹ Read more about the GitHub remote configurations that usethis supports at:
  <https://happygitwithr.com/common-remote-setups.html>.

And I still can only commit but not Push. 

git remote -v
origin  https://github.com/adsmithca/libminer.git (fetch)
origin  https://github.com/adsmithca/libminer.git (push)

Not sure where to go from here without breaking something...

**Comment by ateucher** [comment]
Hmmmm, everything looks good there Andrew. Do you still have the script that has your credentials in it in your folder?
